src/probe/probe.c

```diff
@@ -15,42 +22,529 @@
 #define trace_probe_error(__e, ...) \
 	trace_error(TRACE_CLASS_PROBE, __e, ##__VA_ARGS__)
 
+#define PROBE_DMA_INVALID	0xFFFFFFFF
+#define PROBE_POINT_INVALID	0xFFFFFFFF
+
+#define PROBE_BUFFER_LOCAL_SIZE		8192
+
+struct probe_dma_buf {
+	uintptr_t w_ptr;
+	uintptr_t r_ptr;
+	uintptr_t addr;
+	uintptr_t end_addr;
+	uint32_t size;
+	uint32_t avail;
+};
+
+struct probe_dma_ext {
+	uint32_t stream_tag;
+	uint32_t dma_buffer_size;
+	struct dma_sg_config config;
+	struct probe_dma_buf dmapb;
+	struct dma_copy dc;
+};
+
+struct probe_pdata {
+	struct probe_dma_ext ext_dma;
+	struct probe_dma_ext inject_dma[CONFIG_PROBE_DMA_MAX];
+	struct probe_point probe_points[CONFIG_PROBE_POINTS_MAX];
+};
+
+static struct probe_pdata *_probe;
```

Comment:
can you align with @tlauda , he is optimising usage of data section and heap.

Reply:
Could you please describe what do you mean exactly by these optimizations?

Reply:
@abonislawski he was making changes for SMP IIRC, best chat with him or have him ACK this.

Reply:
@lgirdwood probe_pdata added to sof context, so we're good. In the future we can think whether sharing is needed.

Reply:
@tlauda ok great, @abonislawski just need the comments and were good.